src/models/visualisation.py

```diff
@@ -0,0 +1,213 @@
+            Defaults to CHERNOBYL_COORDS_WGS84.
+        crs (str, optional): coordinates reference system to be used.
+            Defaults to UTM35N.
+        add_layer_control (bool, optional): whether to add layer controls to map.
+            Warning: only use this when you don't intend to add any additional data
+            after calling this function to the map. May cause bugs otherwise.
+            Defaults to False.
+
+    Returns:
+        folium.Map: map with polygons and graph displayed as described
+    """
+
+    if folium_tile_list is None:
+        folium_tile_list = ["OpenStreetMap"]
+
+    if m is None:
+        m = folium.Map(location, zoom_start=8, tiles=folium_tile_list.pop(0))
+
+    # Adding standard folium raster tiles
+    for tiles in folium_tile_list:
+        # special esri satellite data case
+        if tiles == "esri":
+            folium.TileLayer(
+                tiles=(
+                    "https://server.arcgisonline.com/ArcGIS/rest/"
+                    "services/World_Imagery/MapServer/tile/{z}/{y}/{x}"
+                ),
+                attr="esri",
+                name="esri satellite",
+                overlay=False,
+                control=True,
+            ).add_to(m)
+        else:
+            folium.TileLayer(tiles=tiles).add_to(m)
+
+    # Adding polygon data
+    if polygon_gdf is not None:
+        # creating a color index that maps each category
+        # in the color_column to an integer
+        polygon_gdf["index"] = polygon_gdf.index
+        polygon_gdf["color_index"] = (
+            polygon_gdf[color_column].astype("category").cat.codes.astype("int64")
+        )
+
+        choropleth = folium.Choropleth(
+            polygon_gdf,
+            data=polygon_gdf,
+            key_on="feature.properties.index",
+            columns=["index", "color_index"],
+            fill_color="YlOrBr",
+            name=name + "_polygons",
+        )
+        choropleth.add_to(m)
+
+        # adding popup markers with class name
+        folium.features.GeoJsonPopup(fields=[color_column], labels=True).add_to(
+            choropleth.geojson
+        )
+
+    # Adding graph data
+    if graph is not None:
+        node_gdf, edge_gdf = create_node_edge_geometries(graph, crs=crs)
+
+        # add graph edges to map
+        if not edge_gdf.empty:
+            edges = folium.features.GeoJson(
+                edge_gdf,
+                name=name + "_graph_edges",
+                style_function=get_style_function("#dd0000"),
+            )
+            edges.add_to(m)
+
+        # add graph nodes/vertices to map
+        node_marker = folium.vector_layers.Circle(radius=100, color="black")
+        nodes = folium.features.GeoJson(
+            node_gdf, marker=node_marker, name=name + "_graph_vetrices"
+        )
+        nodes.add_to(m)
+
+    if add_layer_control:
+        folium.LayerControl().add_to(m)
+
+    return m
+
+
+def create_node_edge_geometries(
+    G: nx.Graph, crs: str = UTM35N
```

Comment:
nit: variable naming

```suggestion
    graph: nx.Graph, crs: str = UTM35N
```